src/control/server/config.go

```diff
@@ -148,9 +148,9 @@ func saveActiveConfig(config *configuration) {
 // hash produces unique int from string, mask MSB on conversion to signed int
 func hash(s string) int {
 	h := fnv.New32a()
-	h.Write([]byte(s))
-	// mask MSB of uint32 as this will be sign bit
-	return int(h.Sum32() & 0x7FFFFFFF)
+	_, _ = h.Write([]byte(s)) // ignore return values
```

Comment:
Seems like it would be better here to assert that there's no error on write. If there's a "this can't happen" error then a panic() is justified IMO to avoid moving on with life and potentially failing in strange and unexpected ways elsewhere.

Reply:
I left this as all the example code that uses hash.Write in this way ignore the return values. there are different error codes returned so I think it would involve some type assertion on the errors. Introducing error handling in this patch would be wrong IMO as we don't want to be changing behaviour in a linter fix patch. To keep separation of concerns the above maintains the current behaviour and just fixes the warning.

Reply:
While Write returns an int and error, these are just to comply with the io.Writer interface. A hash will never return an error or an n != len(byteSlice).
https://github.com/golang/go/wiki/Hashing